source/grid/tria.cc

```diff
@@ -950,8 +950,9 @@ namespace
     Assert(face_1->at_boundary() && face_2->at_boundary(),
            ExcMessage("Periodic faces must be on the boundary"));
 
+    if (std::abs(cell_1->level() - cell_2->level()) > 1)
+      return;
```

Comment:
This case can only happen in 1d. It might be worth asserting this. But regardless, can you explain why in that case there is nothing to do in the function?

Reply:
See also #8423. I have moved the assert into the else-statement of the above if-else construct!

I must I do not have an explanation for the different behavior of 1D and 2/3D. Do you have an answer?

Reply:
You mean why this can only happen in 1d? In 2d/3d, we enforce the requirement that each edge can only have at most one hanging node, and as a consequence neighboring cells can differ by at most one refinement level. But in 1d, there are no hanging nodes and so we allow neighboring cells to differ by more than one refinement level.

Since this wasn't clear to you, it may also not be clear to others -- maybe it's worthwhile paraphrasing what I wrote above into a comment at the location where you added the code?

Reply:
Thank you for the explanation! I was also thinking about that... It just felt natural that there is always limitations in the level differences. Limitation is useful for the hanging nodes (p4est, space-filling curves, etc.) and for multigrid...

I have slightly modified your comments and added those in PR #8423.

Reply:
Thanks. I agree that there are good reasons to limit the level difference between neighboring cells. We generally do that through the smoothing flags given to the triangulation and enforced upon mesh refinement, but I don't recall whether there is a way to enforce this kind of difference in 1d.